packages/contractkit/src/wrappers/Validators.ts

```diff
@@ -343,6 +354,35 @@ export class ValidatorsWrapper extends BaseWrapper<Validators> {
     this.contract.methods.affiliate
   )
 
+  async getStatus(
+    validator: Address,
+    blocks: any[],
+    electedSigners: Address[],
+    frontRunnerSigners: Address[]
+  ): Promise<ValidatorStatus> {
+    const accounts = await this.kit.contracts.getAccounts()
+    const name = (await accounts.getName(validator)) || ''
+    const signer = await accounts.getValidatorSigner(validator)
+    const electedIndex = electedSigners.map((a) => eqAddress(a, signer)).indexOf(true)
+    const frontRunnerIndex = frontRunnerSigners.map((a) => eqAddress(a, signer)).indexOf(true)
+    const proposedCount = blocks.filter((b) => b.miner === signer).length
+    let signedCount = 0
+    if (electedIndex >= 0) {
+      signedCount = blocks.filter((b) =>
+        bitIsSet(parseBlockExtraData(b.extraData).parentAggregatedSeal.bitmap, electedIndex)
+      ).length
+    }
```

Comment:
If the blocks cross an epoch boundary, this method may incorrectly attribute blocks signed to this validator. (On second look, the same bug exists in `master`)

Reply:
Let's address in a follow-up PR

Reply:
Created an issue

Reply:
https://github.com/celo-org/celo-monorepo/issues/2292